### DISCUSSION TITLE
the modules field in the weblog message for a started event in a DSL2 workflow

### DISCUSSION THREAD

**Discussion by zhangzhen** [nextflow-io/nextflow#1960] [Q&A]
The weblog message contains the "modules" field under the "metadata:parameters" field when a pipeline written in DSL2 is started. The "modules" field lists process names without subworkflow names, and process names in the scope of the main workflow are not shown unexpectedly. However, the log message shown in the shell when launching a pipeline is correct.
How is this issue solved?

**Comment by zhangzhen** [comment]
the "modules" field comes the "modules" key from within my modules.conf.

**Reply by pditommaso** [reply]
The weblog is not actively maintained anymore. Consider using https://tower.nf instead

**Reply by zhangzhen** [reply]
@pditommaso Thank you for your recommendation. I've tried nf-tower and found that it has a nice UI and works very well. However, I love the simplicity of weblog, and weblog also provides what I need. Based upon weblog messages, I implemented a mini-tower with Phoenix/Elixir, which was my practice with the use of the state approach. If I have enough time, I will give building around the Webhook APIs nf-tower provides a try.
